gn2/wqflask/oauth2/toplevel.py

```diff
@@ -81,7 +81,8 @@ def __success__(token):
                 "token": session.user_token(),
                 "logged_in": True
             })
-            return redirect("/")
+            redirect_url = session.session_info().get("redirect_url", "/")
+            return redirect(redirect_url)
```

Comment:
Clear the "redirect_url" value in the session here, just before the `return redirect(redirect_url)` statement. Don't leave the now unnecessary value in place.